third_party/terraform/data_sources/data_source_google_firebase_project.go.erb

```diff
@@ -0,0 +1,56 @@
+<% autogen_exception -%>
+package google
+<% unless version == "ga" -%>
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGoogleFirebaseProject() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceFirebaseProjectRead,
+		Schema: map[string]*schema.Schema{
```

Comment:
We prefer to keep the datasource and resource schema as close as possible so we can use `datasourceSchemaFromResourceSchema` to derive the schema instead of duplicating it.

Reply:
The schemas are actually identical other than "Compute/Optional/Required/Description". I didn't even realize there was a function that handled that.  Do I just use that function to derive the schema or can the entire data source be generated?

Reply:
I don't mind splitting this into a separate PR if you'd prefer since it's really unrelated to adding the location_id.

Reply:
The function is just to derive the schema. But if it makes sense you can also just call through to the "read" method of the resource too. It can keep the datasource very small in terms of code and also updates to the resource such as new fields don't have to be replicated. A example of this is [the compute router datasource](https://github.com/terraform-providers/terraform-provider-google/blob/b658404d0a33cfbe1f5674a32b1b00465f1907ef/google/data_source_google_compute_router.go)